src/Dialect/Krnl/KrnlOps.cpp

```diff
@@ -996,20 +997,23 @@ LogicalResult KrnlCopyFromBufferOp::verify() {
 void KrnlSeqExtractOp::getEffects(
     SmallVectorImpl<SideEffects::EffectInstance<MemoryEffects::Effect>>
         &effects) {
-  effects.emplace_back(MemoryEffects::Read::get(), getSeq(),
+  effects.emplace_back(MemoryEffects::Read::get(), &getSeqMutable(),
       SideEffects::DefaultResource::get());
-  effects.emplace_back(MemoryEffects::Write::get(), getOutput(),
+  effects.emplace_back(MemoryEffects::Read::get(), &getIndexMutable(),
```

Comment:
This was not in the original implementation, but the index is read in this op, so I added the read effect to it.

Reply:
@chentong319 as the implementer of Sequences, please double check its ok 
(Tong away for a bit, so don't stall the PR for this, Tong can check it later and revert if needed)

Reply:
I agree on adding IndexMutable() for its read. But we should keep the original write for Output(). An extract line?
 ```
 effects.emplace_back(MemoryEffects::Write::get(), &getOutputMutable(),
```